### PR TITLE
Upgrade to Go 1.22.4 to fix security vulnerability

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,7 +2,7 @@ module github.com/mongodb/mongodb-atlas-kubernetes/v2
 
 go 1.22.0
 
-toolchain go1.22.3
+toolchain go1.22.4
```
Comment (helderjs) [inline go.mod:5]: What about Dockerfile? why not include the tooling in the update?

**Review by helderjs — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (helderjs) [reply go.mod:5]: this is one more reason to do, no? We "skip" the magic and ensure tooling explicitly reflects what it should be

Comment (josvazg) [reply go.mod:5]: I think this magic is done on purpose by the Go team. Basically means you do not need to wait for whatever distribution channel to get the latest go tool chain version. You do not need to wait for the docker container with go 1.22.4 to be ready, you do not need the GitHub CI ubuntu image to have the latest go tool chain ready, your nix packages or your internal CI system to be up to date.

Comment (josvazg) [reply go.mod:5]: I updated the tools' toolchain the same way.

**Review by helderjs — APPROVED** [review]

**Pull request merged**